pybamm/batch_study.py

```diff
@@ -0,0 +1,51 @@
+#
+# BatchStudy class
+#
+import pickle
+import pybamm
+from itertools import product
+
+
+class BatchStudy:
+    """
+    comparison class
+    """
+
+    def __init__(
+        self, models, solvers=None, experiments=None, repeats=1, permutations=False
+    ):
+
+        self.models = models
+        self.solvers = solvers
+        self.experiments = experiments
+        self.repeats = repeats
+        self.permutations = permutations
+
+    def solve(self):
+        self.sims = []
+        for num in range(self.repeats):
+            if self.permutations is False:
+                for model, solver, experiment in zip(
+                    self.models.values(),
+                    self.solvers.values(),
+                    self.experiments.values(),
+                ):
+                    sim = pybamm.Simulation(model, solver=solver, experiment=experiment)
+                    sim.solve([0, 3600])
+                    self.sims.append(sim)
+            else:
+                for values in product(
+                    self.models.values(),
+                    self.solvers.values(),
+                    self.experiments.values(),
+                ):
+                    model, solver, experiment = values
+                    sim = pybamm.Simulation(model, solver=solver, experiment=experiment)
+                    sim.solve([0, 3600])
+                    self.sims.append(sim)
```

Comment:
repeats should only be called on sim.solve, and the solve times should be recorded and averaged, e.g.:
```python3
def solve(self, t_eval=None):
        self.sims = []
        if self.permutations is False:
            iter = zip(self.models.values(), self.solvers.values(), self.experiments.values())
        else:
            iter = product(self.models.values(), self.solvers.values(), self.experiments.values())
        for model, solver, experiment in zip(self.models.values(), self.solvers.values(), self.experiments.values()):
            sim = pybamm.Simulation(model, solver=solver, experiment=experiment)
            # repeat to get average solve time and integration time
            solve_time = 0
            integration_time = 0
            for num in range(self.repeats):
              sol = sim.solve(t_eval)
              solve_time += sol.solve_time
              integration_time += sol.integration_time
            sim.solution.solve_time = solve_time / self.repeats
            sim.solution.integration_time = integration_time / self.repeats
            self.sims.append(sim)
```